doc/TorEvents.markdown

```diff
@@ -0,0 +1,301 @@
+would occur for a single client connection:
+
+### PRIVCOUNT_DNS_RESOLVED
+
+This event is sent when an exit receives a client request, including:
+* both connect and resolve requests,
+* for DNS names, and IPv4 and IPv6 addresses,
+* whether the DNS name is in Tor's local DNS cache or not,
+* with limited checks on whether the address is a valid DNS name or IP address.
+  (Tor clients and Tor Exits perform some sanity checks before sending an
+  address for resolution).
+
+There is *no* filtering of unsuccessful DNS hostname lookups. However,
+internal tor network requests (such as directory and onion service requests)
+do not use DNS resolution, so they do not trigger this event.
+
+The DNS event is *not* used by any PrivCount counters, but was used as part of
+PrivEx's censorship blacklist measurements.
+
+It includes the following fields:
+* Channel ID
+* Circuit ID
+* Address requested by client (hostname or IPv4 or IPv6)
+
+It has the following known issues:
+* There is no indication in the event whether the request was successful or
+  not, so this event can not be used to count client connections (only client
+  requests).
+  https://github.com/privcount/privcount/issues/184
+* The resolved address is not included in the event.
+  https://github.com/privcount/privcount/issues/184
+* The time is not included in the event.
+  https://github.com/privcount/privcount/issues/187
+* These events include relay DirPort self-checks to their own IPv4 addresses
+  https://github.com/privcount/privcount/issues/188
+
+### PRIVCOUNT_STREAM_BYTES_TRANSFERRED
+
+This event is sent when tor reads or writes data from a remote exit stream.
+
+Internal tor network requests (such as onion service requests) do not trigger
+this event: they are filtered out on the tor side. Zero reads and writes do
+not trigger this event. Tor also performs some checks before reading or
+writing that may cause this event not to be sent.
+
+Client BEGINDIR (ORPort directory) requests trigger this event, but client
+HTTP (DirPort directory) requests do not.
+
+Relay DirPort self-checks by remote relays do trigger this event, even though
+they are not client traffic. The self-testing traffic and connections are
+neglible compared with all tor network traffic, but may be significant for
+small counters that include IPv4 ports 80 or 9030.
+
+Some tor-side filtering of this event may be necessary for performance
+reasons: this event is the most frequent event that tor sends to PrivCount.
+
+The bytes event is used by the PrivCount traffic model counters. (Other
+counters that report bytes use the same data as this event, but aggregate it,
+and send the totals with the END event.)
+
+It includes the following fields:
+* Channel ID (if the OR channel has not been cleared)
+* Circuit ID (if there is a corresponding OR circuit)
+* Stream ID
+* Is Outbound Flag
+* Number of Bytes
+* Current Timestamp
+
+It has the following known issues:
```

Comment:
With our traffic model counting code, we are trying to model *all* traffic flowing between relays. So I think we want to include BEGINDIR requests, right? Perhaps not DirPort self-checks though...